backend-api/src/functions/asyncActiveSession/tokenService/publicKeyGetter.ts

```diff
@@ -43,18 +53,73 @@ export class PublicKeyGetter implements IPublicKeyGetter {
       publicKey = await importJWK(jwk, jwk.alg);
     } catch (error) {
       return errorResult({
-        errorMessage: `Error converting JWK to key object: ${error}`,
-        errorCategory: "CLIENT_ERROR",
+        errorMessage: `Error getting public key: ${error}`,
+        errorCategory: "SERVER_ERROR",
       });
     }
 
     return successResult(publicKey);
   }
+
+  private getTokenKid(jwt: string): string {
+    const decodedProtectedHeader: ProtectedHeaderParameters =
+      decodeProtectedHeader(jwt);
+    const { kid } = decodedProtectedHeader;
+    if (!kid) {
+      throw new Error("kid not present in JWT header");
+    }
+    return kid;
+  }
+
+  private async getJwk(jwksEndpoint: string, kid: string): Promise<IJwk> {
+    const response = await this.sendHttpRequest(
```

Comment:
Send HTTP request to STS here in the PublicKeyGetter class instead of in the TokenService